Search bar

2001-hagar opened the pull request.

Commit:
search-bar

Commit:
enter-event

Ahmed-Khaled24 commented on src/search-bar/search.jsx:
```diff
@@ -0,0 +1,25 @@
+
+const Search = () => {
+  
+      const eventkey = (event) => {
```
we can use a better name for this function to make it easier to read.

Ahmed-Khaled24 commented on src/search-bar/search.jsx:
```diff
@@ -0,0 +1,25 @@
+
+
+
+const Search = () => {
```
to use this component I think we will have to pass a `value` and `onChange` props to store the input value and handle user typing on this component.

Ahmed-Khaled24 commented on src/search-bar/search.css:
```diff
@@ -0,0 +1,64 @@
+@import 
+url('https://fonts.googleapis.com/css2?family=Rubik:ital,wght@0,300..900;1,300..900&display=swap');
+
+.logo{
```
not mandatory but using a very general class name makes the code harder to read.

Ahmed-Khaled24 commented on src/search-bar/search.css:
```diff
@@ -0,0 +1,64 @@
+
+}
+
+.text{
```
not mandatory but using a very general class name makes the code harder to read.

Commit:
adding prop

Ahmed-Khaled24 commented on src/components/search-bar/search.component.jsx:
```diff
@@ -0,0 +1,24 @@
+  return (
+     <div className='search-bar-container'>
+     <img src={searchlogo} className="logo-search" alt="search logo" />
+     <input type='text' id="Text" className="text-search" value={searchinput} placeholder="Search" onKeyDown={(e) => eventkey(e)}/>
```
so now we have access to the value of the input, we need to be able to change it when the input field changes.

Ahmed-Khaled24 commented on src/components/search-bar/search.component.jsx:
```diff
@@ -0,0 +1,24 @@
+const SearchBar = ({searchinput}) => {
+  
+
+      const eventkey = (event) => {
+      if(event.keyCode === 13){
+        document.getElementById("Text").classList.add("enteredtext");
+        setTimeout(function(){document.getElementById("Text").classList.remove("enteredtext")
+      },500)
+    }      
+    }
```
- the function name is very general and doesn't give any hint about what does it do for us.
- manipulating the DOM this way is not recommended in React you may use `useRef` or create a piece of state to handle this and add the class name conditionally.

Commit:
feat: handlechange-function which keeps track the change in the input of the search-bar Search-Bar featurer/handlechange-function

perf: creating a state to add the class name conditionally instead of using DOM Search-Bar perf/isEntered-state

Ahmed-Khaled24 commented:
Merged locally then pushed to fix the merge conflict, Good job👍🏼
